q2_metadata/tests/test_merge.py

```diff
@@ -201,3 +197,53 @@ def test_merge_no_samples_or_columns_overlapping(self):
                                   'col4', 'col5', 'col6']))
 
         self.assertEqual(obs1, exp1)
+
+    def test_invalid_index_name_in_error_message(self):
```

Comment:
Let's rename this test something a bit different - I wasn't immediately sure what 'invalid index name' meant. This is testing the behavior for merging two metadata files with mismatched ID columns. With this in mind, try to come up with a name that tells the reader what the exact use-case is that you're testing!